test/e2e/e2e_go_cluster_test.go

```diff
@@ -0,0 +1,177 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Modified from https://github.com/kubernetes-sigs/kubebuilder/tree/39224f0/test/e2e/v3
+
+package e2e_test
+
+import (
+	"encoding/base64"
+	"fmt"
+	"path/filepath"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo" //nolint:golint
+	. "github.com/onsi/gomega" //nolint:golint
+	kbtestutils "sigs.k8s.io/kubebuilder/test/e2e/utils"
+)
+
+var _ = Describe("operator-sdk", func() {
+	var controllerPodName string
+
+	Context("built with operator-sdk", func() {
+		BeforeEach(func() {
+			By("deploying project on the cluster")
+			err := tc.Make("deploy", "IMG="+tc.ImageName)
+			Expect(err).Should(Succeed())
+		})
+		AfterEach(func() {
+			By("deleting Curl Pod created")
+			_, _ = tc.Kubectl.Delete(true, "pod", "curl")
+
+			By("deleting CR instances created")
+			sampleFile := filepath.Join("config", "samples",
+				fmt.Sprintf("%s_%s_%s.yaml", tc.Group, tc.Version, strings.ToLower(tc.Kind)))
+			_, _ = tc.Kubectl.Delete(false, "-f", sampleFile)
+
+			By("cleaning up permissions")
+			_, _ = tc.Kubectl.Command("delete", "clusterrolebinding",
+				fmt.Sprintf("metrics-%s", tc.TestSuffix))
+
+			By("undeploy project")
+			_ = tc.Make("undeploy")
+		})
+
+		It("should run correctly in a cluster", func() {
+			By("checking if the Operator project Pod is running")
+			verifyControllerUp := func() error {
+				By("getting the controller-manager pod name")
+				podOutput, err := tc.Kubectl.Get(
+					true,
+					"pods", "-l", "control-plane=controller-manager",
+					"-o", "go-template={{ range .items }}{{ if not .metadata.deletionTimestamp }}{{ .metadata.name }}"+
+						"{{ \"\\n\" }}{{ end }}{{ end }}")
+				Expect(err).NotTo(HaveOccurred())
+
+				By("ensuring the created controller-manager Pod")
+				podNames := kbtestutils.GetNonEmptyLines(podOutput)
+				if len(podNames) != 1 {
```

Comment:
So, shouldn't most of these conditionals be using ginkgo-style checks? Like this should be Expect(len(podNames)).To(Equal(1)) instead of manually bubbling up an error?

Reply:
Since the other tests are with the same check. I think it would be nice be done horizontally to all after we get Ansible merged as well. WDYT?

Reply:
See; https://github.com/operator-framework/operator-sdk/issues/3771

Reply:
:+1:

Reply:
Cross-post from https://github.com/operator-framework/operator-sdk/issues/3771#issuecomment-691482813:

> I think this can be improved further to `Expect(podNames).To(HaveLen(1))`